src/main/scala/devices/tilelink/CanHaveBuiltInDevices.scala

```diff
@@ -11,22 +11,30 @@ trait HasBuiltInDeviceParams {
   val errorDevice: Option[DevNullParams]
 }
 
+sealed trait BuiltInDevices {
```

Comment:
I the main difference I care about vs a final case class is that the case class generates a bunch of methods and extends stuff that I don't think are appropriate/useful. like case class extends `serializable` which `TLError` and `TLZero` definitely are not and it generates stuff like unapply that I don't want exposed because that API will probably be broken.

sealed is for similar reasons. to prevent anyone else from creating `BuiltInDevices` outside of `attachBuiltInDevices` and just narrow down the API surface area.

just my personal preference, maybe I'm being a too cautious or this isn't the right approach? open to changing this.

Reply:
> just my personal preference, maybe I'm being a too cautious or this isn't the right approach? open to changing this.

No, I think it's a good idea to limit API surface area. I was mostly wondering `sealed trait` vs `final case class` and not `sealed trait` vs `(not sealed) trait`. I hadn't thought of the extra methods and traits that case classes provide as being extra stuff that consumers could possibly depend on that would become later liabilities, but now that you point it out, I agree, especially because this data structure is, at the moment, an arbitrary bundle of "stuff", and we may want to reserve the right to change this in the future.